docs/modules/spark-k8s/pages/usage-guide/job-dependencies.adoc

```diff
@@ -93,28 +72,34 @@ With this method, the job dependencies are provisioned from a `PersistentVolume`
 include::example$example-sparkapp-pvc.yaml[]
 ----
 <1> Job artifact located on S3.
-<2> Job main class
-<3> Spark dependencies: the credentials provider (the user knows what is relevant here) plus dependencies needed to access external resources (in this case, in s3, accessed without credentials)
-<4> the name of the volume mount backed by a `PersistentVolumeClaim` that must be pre-existing
-<5> the path on the volume mount: this is referenced in the `sparkConf` section where the extra class path is defined for the driver and executors
+<2> Name of the main class to run.
+<3> The job dependencies provisioned from the volume below are added to the class path of the driver and executors.
+<4> A `PersistentVolumeClaim` created by the user prior to submitting the Spark job.
+<5> The volume containing the dependencies is mounted in the job pod.
+<6> The volume containing the dependencies is mounted in the driver pod.
+<7> The volume containing the dependencies is mounted in the executor pods.
 
 NOTE: The Spark operator has no control over the contents of the dependency volume. It is your responsibility to make sure all required dependencies are installed in the correct versions.
 
-A `PersistentVolumeClaim` and the associated `PersistentVolume` can be defined like this:
+A `PersistentVolumeClaim` and the associated `PersistentVolume` can be defined and provisioned like this:
 
 [source,yaml]
 ----
 include::example$example-pvc.yaml[]
 ----
-<1> Reference to a `PersistentVolume`, defining some cluster-reachable storage
-<2> The name of the `PersistentVolumeClaim` that references the PV
-<3> Defines a `Volume` backed by the PVC, local to the Custom Resource
-<4> Defines the `VolumeMount` that is used by the Custom Resource
-
+<1> Create a volume. This definition, the size and type of the volume are highly dependent on the type of cluster you are using.
+<2> Create a persistent volume claim. This allows the volume to be populated with the necessary dependencies and later on referenced by the Spark job.
+<3> The volume name is referenced by the `PersistentVolumeClaim`.
+<4> Create a job that mounts the volume and populates it with the necessary dependencies. This must job can be run before submitting the Spark job.
+<5> The job references the `PersistentVolumeClaim` created above.
 
 === Maven packages
 
 The last and most flexible way to provision dependencies is to use the built-in `spark-submit` support for Maven package coordinates.
+The downside of this method is that job dependencies are downloaded every time the job is submitted and this has several implications you must be aware of.
+For example, the job submission time will be longer than with the other methods
+Network connectivity problems may lead to job submission failures.
+And finally, not all type of dependencies can be provisioned this way. Most notably, JDBC drivers cannot be provisioned this way since the JVM will only look for them at startup time.
```

Comment:
```suggestion
And finally, not all type of dependencies can be provisioned this way.
Most notably, JDBC drivers cannot be provisioned this way since the JVM will only look for them at startup time.
```